test/impl/AzureBlobFiles.test.js

```diff
@@ -571,6 +606,41 @@ describe('_getPresignUrl', () => {
     expect(url).toEqual(expectedUrl)
     expect(tvm.getAzureBlobPresignCredentials).toHaveBeenCalledWith({ blobName: 'fakesub/afile', expiryInSeconds: 60, permissions: 'fake' })
   })
+
+  test('_getPresignUrl with correct options default permission own credentials', async () => {
```

Comment:
the 4 tests should not simulate an init by setting private control variables. Instead those variable should be set by the constructor depending on the init input.
for example instead of setting `files.hasOwnCredentials=true` and `files.credentials=fakeUserCredentials` and `files._azure.sasCreds=false` you could do `files = await AzureBlobFiles.init(fakeUserCredentials, null)` => this will do the same but doesn't test internal control variable names which could change in the future